packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2676,4 +2682,25 @@ export class TransactionController extends BaseControllerV1<
       log('Error while updating post transaction balance', error);
     }
   }
+
+  private async publishTransactionWithValidation(
+    rawTx: string,
+    transactionMeta: TransactionMeta,
+  ): Promise<string> {
+    let hash;
+    try {
+      hash = await this.publishTransaction(rawTx);
+      // TODO: Replace `any` with type
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      if (error?.code === errorCodes.rpc.invalidInput) {
```

Comment:
Is it worth us checking both the code itself but also the message content to cast a wider net?

Reply:
Done in https://github.com/MetaMask/core/pull/3800/commits/c68f7fb7014444efe0ea97fd3471f937489e6871

Reply:
Done in https://github.com/MetaMask/core/pull/3800/commits/dc0a169ff0dc7574f1537ff6e67e57eb0bf0e08d